lib/components/narrative/default/default-itinerary.js

```diff
@@ -2,50 +2,179 @@ import coreUtils from '@opentripplanner/core-utils'
+          <div style={{
+            width: '20px',
+            height: '20px',
+            position: 'absolute',
+            display: 'inline',
+            paddingLeft: '4px',
+            paddingBottom: '4px'
+          }}>
+            <LegIcon leg={leg} />
+          </div>
+        </span>
+      )
+    }
+  }
+]
+
 export default class DefaultItinerary extends NarrativeItinerary {
+  _onMouseOver = () => {
+    const {index, setVisibleItinerary} = this.props
+    if (typeof setVisibleItinerary === 'function') setVisibleItinerary({index})
+  }
+
+  _onMouseOut = () => {
+    const {setVisibleItinerary} = this.props
+    if (typeof setVisibleItinerary === 'function') setVisibleItinerary({index: null})
+  }
+
+  _isSortingOnAttribute = (attribute) => {
+    const {sort} = this.props
+    if (sort && sort.type) {
+      const type = sort.type.toLowerCase()
+      return attribute.id.toLowerCase() === type ||
+        (attribute.alias && attribute.alias.toLowerCase() === type)
+    }
+    return false
+  }
+
   render () {
     const {
       active,
-      activeLeg,
-      activeStep,
       expanded,
-      index,
       itinerary,
       LegIcon,
-      setActiveLeg,
-      setActiveStep
+      timeFormat
     } = this.props
+    const timeOptions = {
+      format: timeFormat,
+      offset: coreUtils.itinerary.getTimeZoneOffset(itinerary)
+    }
     return (
-      <div className={`option default-itin${active ? ' active' : ''}`}>
+      <div
+        className={`option default-itin${active ? ' active' : ''}${expanded ? ' expanded' : ''}`}
+        role='presentation'
+        // FIXME: Move style to css
+        style={{
+          borderLeft: active && !expanded ? '4px teal solid' : undefined,
+          backgroundColor: expanded ? 'white' : undefined
+        }}
+        onBlur={this._onMouseOut}
+        onFocus={this._onMouseOver}
+        onMouseOut={this._onMouseOut}
+        onMouseOver={this._onMouseOver}
+      >
         <button
           className='header'
+          // _onHeaderClick comes from super component (NarrativeItinerary).
           onClick={this._onHeaderClick}
         >
-          <span className='title'>Itinerary {index + 1}</span>{' '}
-          <span className='duration pull-right'>{formatDuration(itinerary.duration)}</span>{' '}
-          <span className='arrivalTime'>{formatTime(itinerary.startTime)}—{formatTime(itinerary.endTime)}</span>
+          <div
+            className='title'>
+            {getItineraryDescription(itinerary)}
+          </div>
+          <ul className='list-unstyled itinerary-attributes'>
+            {ITINERARY_ATTRIBUTES
+              .sort((a, b) => {
+                const aSelected = this._isSortingOnAttribute(a)
+                const bSelected = this._isSortingOnAttribute(b)
+                if (aSelected) return -1
+                if (bSelected) return 1
+                else return a.order - b.order
+              })
+              .map(attribute => {
```

Comment:
The bikeshare result for the query below show the "main" activity icon to be "Walk":

![image](https://user-images.githubusercontent.com/56846598/88728556-0312f280-d100-11ea-9b32-d3d10b94339c.png)

http://localhost:9966/#/?ui_activeSearch=pxn1jrumg&ui_activeItinerary=1&fromPlace=600%20Columbia%20St%2C%20Orlando%2C%20FL%2C%20USA%3A%3A28.527074880317347%2C-81.38654708862306&toPlace=2521%20Dade%20Avenue%2C%20Orlando%2C%20FL%2C%20USA%3A%3A28.574572543440226%2C-81.37418746948244&date=2020-07-28&time=18%3A15&arriveBy=false&mode=WALK%2CBUS%2CRAIL&showIntermediateStops=true&maxWalkDistance=1207&optimize=QUICK&walkSpeed=1.34&ignoreRealtimeUpdates=true&companies=&bannedRoutes=&numItineraries=3&otherThanPreferredRoutesPenalty=900&preferredRoutes=

Reply:
I think we should just leave a FIXME for now. There are some unresolved design questions around this. Do we want this to unconditionally show walk time if there is a walking leg? Or should it show combined active time?